processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -159,8 +159,18 @@ public GroupByQueryQueryToolChest(
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
       final List<AggregatorFactory> aggs = Lists.newArrayList();
+
       for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
```

Comment:
@himanshug I gave that query a try and everything seemed to work as expected. 

@gianm The outer query's aggregator specs are not actually thrown away. I think what's happening is that 1) the inner query is executed using the inner query's agg specs, then 2) the sequence is written into an intermediate incremental index, and finally 3) the outer query is executed against that intermediate incremental index. The fudged list of agg specs isn't used for (1) or (3) but is used to construct the incremental index in (2) to make sure that, according to the comment in the code, the inner incremental index has all the columns required by the outer query.

Having said that, I was thinking there might be a better way to do this - use the inner query's agg specs to build the intermediate incremental index instead of the outer query's one, and then throw an exception if there are any aggregators in the outer query that have a fieldName that doesn't match an aggregator 'name' from the inner query. This would change the behavior from allowing the request but returning 0 for the outer aggregator that referenced a non-existent aggregator in the inner query to failing the query with an exception. Is this okay? 

Here's an example of current behavior for reference:
```
{
  "queryType": "groupBy",
  "dataSource": {
    "type": "query",
    "query": {
      "queryType": "groupBy",
      "dataSource": "wikipedia",
      "granularity": "all",
      "dimensions": [
        
      ],
      "aggregations": [
        {
          "type": "longSum",
          "fieldName": "count",
          "name": "inner_1"
        }
      ],
      "intervals": [
        "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
      ]
    }
  },
  "granularity": "hour",
  "dimensions": [
    
  ],
  "aggregations": [
    {
      "type": "longSum",
      "fieldName": "inner_1",
      "name": "outer_1"
    },
    {
      "type": "longSum",
      "fieldName": "inner_non_existent",
      "name": "outer_2"
    }
  ],
  "intervals": [
    "2012-01-01T00:00:00.000\/2020-01-03T00:00:00.000"
  ]
}
```
Output:
```
[ {
  "version" : "v1",
  "timestamp" : "2012-01-01T07:00:00.000Z",
  "event" : {
    "outer_1" : 2603,
    "outer_2" : 0
  }
} ]
```